File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    length = len(numbers)
    def dfs(index, value):
        nonlocal answer
         
        if value == target and index == (length - 1):
            answer += 1
        
        next_index = index + 1
        if next_index < length:
            dfs(next_index, value + numbers[next_index])
            dfs(next_index, value - numbers[next_index])
            
    dfs(0, numbers[0])
    dfs(0, -numbers[0])
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let answer = 0;
    const DFS = (total, index) => {       
        if (index === numbers.length) {
            if (total === target) {
                answer++
            }
            return
        }
        
        DFS(total + numbers[index], index + 1)
        DFS(total - numbers[index], index + 1)
    }
    
    DFS(0, 0)
    
    return answer;
}